guide/src/migration.md

```diff
@@ -246,7 +246,7 @@ For example, the following APIs have gained updated variants:
 
 Because the new `Bound<T>` API brings ownership out of the PyO3 framework and into user code, there are a few places where user code is expected to need to adjust while switching to the new API:
 - Code will need to add the occasional `&` to borrow the new smart pointer as `&Bound<T>` to pass these types around (or use `.clone()` at the very small cost of increasing the Python reference count)
-- `Bound<PyList>` and `Bound<PyTuple>` cannot support indexing with `list[0]`, you should use `list.get_item(0)` instead.
+- `Bound<PyList>`, `Bound<PyTuple>`, and `Bound<PyBytes>` cannot support indexing with `list[i]` or `tuple[i]`, because `std::ops::Index` requires reference return values. Use `list.get_item(i)` and `tuple.get_item(i)` instead. For bytes, use `bytes.as_bytes()[i]` to index the byte slice.
```

Comment:
Now that we have the `Index` impl for `Bound<PyBytes>` it can be removed from here again

Reply:
Ufff I started writing this when I noticed it could be implemented; I didn't mean to commit this 🤦